tgui/packages/tgui/interfaces/common/WirePanel/index.tsx

```diff
@@ -0,0 +1,190 @@
+      { (control_lights & WirePanelControls.WIRE_CONTROL_TRIGGER) && <Box className="indicator_label">{WirePanelControlLabels[WirePanelControls.WIRE_CONTROL_TRIGGER]}: {hacked_controls & WirePanelControls.WIRE_CONTROL_TRIGGER && "On" || "Off"} </Box>}
+      { (control_lights & WirePanelControls.WIRE_CONTROL_RECEIVE) && <Box className="indicator_label">{WirePanelControlLabels[WirePanelControls.WIRE_CONTROL_RECEIVE]}: {hacked_controls & WirePanelControls.WIRE_CONTROL_RECEIVE && "On" || "Off"} </Box>}
+      { (control_lights & WirePanelControls.WIRE_CONTROL_TRANSMIT) && <Box className="indicator_label">{WirePanelControlLabels[WirePanelControls.WIRE_CONTROL_TRANSMIT]}: {hacked_controls & WirePanelControls.WIRE_CONTROL_TRANSMIT && "On" || "Off"} </Box>}
+    </Box>
+  );
+};
+
+// remote silicon wire checks
+const RemoteAccessButton = (props, context) => {
+  const {
+    disabled,
+    ...rest
+  } = props;
+  const { data } = useBackend<WirePanelData>(context);
+  const { is_silicon_user, hacked_controls, is_accessing_remotely, cover_status } = data.wire_panel;
+  let shouldDisable = false;
+  if (is_silicon_user) {
+    if ((hacked_controls & WirePanelControls.WIRE_CONTROL_SILICON)) {
+      if (is_accessing_remotely) {
+        shouldDisable = true;
+      } else {
+        shouldDisable = cover_status !== WirePanelCoverStatus.WPANEL_COVER_OPEN;
+      }
+    }
+  } else {
+    shouldDisable = cover_status !== WirePanelCoverStatus.WPANEL_COVER_OPEN;
+  }
+  return (
+    <Button
+      className="WirePanel-wires-generic-button"
+      disabled={shouldDisable}
+      {...rest}
+    />
+  );
+};
+
+// visually block the entire TGUI window based on silicon control
+export const RemoteAccessBlocker = (is_accessing_remotely: BooleanLike, hacked_controls: number) => {
+  if (!!is_accessing_remotely
+    && (hacked_controls & WirePanelControls.WIRE_CONTROL_SILICON)) {
+    return (
+      <Dimmer fillPositionedParent>
+        <Box
+          verticalAlign="top"
+          textAlign="center"
+          fontSize={2.5}
+          fontFamily="Courier"
+          bold
+          color="red"
+        >
+          <Box className="fa-stack" fontSize={2} mr={1} height={7}>
+            <Icon name="wifi" className="fa-stack-1x" color="blue" />
+            <Icon name="ban" className="fa-stack-2x WirePanel-silicon-disabled" />
+          </Box>
+          <Box inline>REMOTE SILICON<br />ACCESS DISABLED</Box>
+          <Box className="fa-stack" fontSize={2} ml={1} height={7}>
+            <Icon name="wifi" className="fa-stack-1x" color="blue" />
+            <Icon name="ban" className="fa-stack-2x WirePanel-silicon-disabled" />
+          </Box>
+        </Box>
+      </Dimmer>
+    );
+  }
+};
+
+// collapsible wire panel to add into your existing TGUI
+export const WirePanelCollapsible = (props, context) => {
```

Comment:
Nice idea, though not sure *collapsible* wire panels are desired? Probably better to have a layer that determines disabled state properly and let people include that if they want (as that is more likely to be useful as it would fit in e.g. a tab).